test/src/unittests/spectral/test_logspectrum.py

```diff
@@ -109,30 +308,43 @@ def testRegression(self, frameSize=8192 + 1):
     def testZero(self):
         # Inputting zeros should return zero. Try with different sizes
         size = 1024
-        while (size >= 256 ):
-            self.assertEqualVector(LogSpectrum(frameSize = size)(zeros(size))[0], zeros(256))
-            size = int(size/2)
+        while size >= 256:
+            self.assertEqualVector(
+                LogSpectrum(frameSize=size)(zeros(size))[0], zeros(256)
+            )
+            size = int(size / 2)
 
     def testInvalidInput(self):
         self.assertComputeFails(LogSpectrum(), [])
         self.assertComputeFails(LogSpectrum(), [0.5])
 
-
     def testInvalidParam(self):
-        self.assertConfigureFails(LogSpectrum(), { 'frameSize': 1 })
-        self.assertConfigureFails(LogSpectrum(), { 'sampleRate': 0 })
-        self.assertConfigureFails(LogSpectrum(), { 'rollOn': -1})
+        self.assertConfigureFails(LogSpectrum(), {"frameSize": 1})
+        self.assertConfigureFails(LogSpectrum(), {"sampleRate": 0})
+        self.assertConfigureFails(LogSpectrum(), {"rollOn": -1})
+        self.assertConfigureFails(LogSpectrum(), {"nOctaves": 0})
+        self.assertConfigureFails(LogSpectrum(), {"nOctaves": 10})
+
+    def testNOctavesParam(self):
```

Comment:
This tests zero input for different number of octaves, so we can change the name of this test to better represent it. Possibly better to merge with `testZero` to make it cover all considered cases?